src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -18,9 +19,13 @@ internal sealed partial class JsonClassInfo
         // The length of the property name embedded in the key (in bytes).
         private const int PropertyNameKeyLength = 6;
 
-        private readonly List<PropertyRef> _propertyRefs = new List<PropertyRef>();
+        // The limit to how many property names from the JSON are cached before using PropertyCache.
+        private const int PropertyNameCountCacheThreshold = 64;
 
-        // Cache of properties by first ordering. Use an array for highest performance.
+        // The properties on a POCO keyed on property name.
+        public volatile Dictionary<string, JsonPropertyInfo> PropertyCache;
```

Comment:
I'm unclear on what benefit there is to putting volatile on dictionary and array members? If we're trying to be thread safe I don't think this will be of any benefit?

Reply:
This could be is read\written by multiple threads during warm-up. The instance is always replaced entirely - not members added later.